File: cookblog-frontend/src/app/api/resources/category.resource.ts
import { RecipeResource } from './recipe.resource';

export class CategoryResource {
  private readonly _id: number | null;
  private readonly _name: string;
  private readonly _recipes: RecipeResource[] | null;

  constructor(
    id: number | null,
    name: string,
    recipes: RecipeResource[] | null,
  ) {
    this._id = id;
    this._name = name;
    this._recipes = recipes;
  }

  get id() {
    return this._id;
  }

  get name() {
    return this._name;
  }

  get recipes() {
    return this._recipes;
  }
}

File: cookblog-backend/src/main/java/com/example/cookblog/dto/resources/CategoryResource.java
package com.example.cookblog.dto.resources;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Null;
import jakarta.validation.constraints.Size;
import lombok.Builder;
import lombok.extern.jackson.Jacksonized;

import java.util.List;

@Builder
@Jacksonized
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
public record CategoryResource(
        @Null Long id,
        @NotBlank @Size(max = 255) String name,
        @Null @JsonManagedReference List<RecipeResource> recipes
) {

}